Remove code that we don't use to avoid installing additional dependencies.
--- a/ruby/Rakefile
+++ b/ruby/Rakefile
@@ -1,6 +1,4 @@
 require "rubygems"
-require "rubygems/package_task"
-require "rake/extensiontask" unless RUBY_PLATFORM == "java"
 require "rake/testtask"
 
 spec = Gem::Specification.load("google-protobuf.gemspec")
@@ -39,45 +37,6 @@
   end
 end
 
-if RUBY_PLATFORM == "java"
-  if `which mvn` == ''
-    raise ArgumentError, "maven needs to be installed"
-  end
-  task :clean do
-    system("mvn --batch-mode clean")
-  end
-
-  task :compile do
-    system("mvn --batch-mode package")
-  end
-else
-  Rake::ExtensionTask.new("protobuf_c", spec) do |ext|
-    ext.ext_dir = "ext/google/protobuf_c"
-    ext.lib_dir = "lib/google"
-    ext.cross_compile = true
-    ext.cross_platform = [
-      'x86-mingw32', 'x64-mingw32',
-      'x86_64-linux', 'x86-linux',
-      'universal-darwin'
-    ]
-  end
-
-  task 'gem:windows' do
-    require 'rake_compiler_dock'
-    RakeCompilerDock.sh "bundle && IN_DOCKER=true rake cross native gem RUBY_CC_VERSION=2.5.0:2.4.0:2.3.0:2.2.2:2.1.6:2.0.0"
-  end
-
-  if RUBY_PLATFORM =~ /darwin/
-    task 'gem:native' do
-      system "rake genproto"
-      system "rake cross native gem RUBY_CC_VERSION=2.5.0:2.4.0:2.3.0:2.2.2:2.1.6:2.0.0"
-    end
-  else
-    task 'gem:native' => [:genproto, 'gem:windows']
-  end
-end
-
-
 # Proto for tests.
 genproto_output << "tests/generated_code.rb"
 genproto_output << "tests/test_import.rb"
@@ -95,9 +54,6 @@
   sh "rm -f #{genproto_output.join(' ')}"
 end
 
-Gem::PackageTask.new(spec) do |pkg|
-end
-
 Rake::TestTask.new(:test => :build) do |t|
   t.test_files = FileList["tests/*.rb"].exclude("tests/gc_test.rb")
 end
